### ISSUE TITLE
Add documentation to the actual code

### ISSUE THREAD

**Issue by Blackclaws** [danielgerlag/workflow-core#1308]
**Is your feature request related to a problem? Please describe.**
First of all thank you for even creating the library. I couldn't help but notice unfortunately that the usage of this library is mainly documented in samples and some ReadTheDocs but not a lot of documentation about any of the methods exists in the actual code. This makes working with the library more difficult than it has to be because one essentially needs to dig into the source code to understand what some things actually do.

CancelCondition is one example. I expected its a condition that is checked once at its point in the workflow whether the workflow should cancel at this point, seems it is checked regularly throughout the workflow. But can also be used to cancel parallel workflows as per some release notes?

**Describe the solution you'd like**
Add documentation to at least all the public facing methods that describe properly what they do and also add an example usage in there.

**Describe alternatives you've considered**
Trawl through the source code to gather this information myself. I've done this now and its really not efficient.

**Comment by Daafjuuh** [comment]
Agreed! I would like to know that status of things. when I for example sql query a step with a certain workflowId -> status 3 gets returned... but what does this 3 mean? 

Just another example adding to your point :)
